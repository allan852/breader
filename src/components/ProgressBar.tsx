import { Progress } from 'antd'
import React, { PureComponent } from 'react'

interface IProgressBarProps {
    className?: string
    start?: number
    max: number
    time: number
    onEnd?: () => any
}
interface IProgressBarState {
    progress: number
}

class ProgressBar extends PureComponent<IProgressBarProps, IProgressBarState> {
    public constructor(props: IProgressBarProps) {
        super(props)
        this.state = { progress: props.start || 0 }
    }
    public componentDidMount () {
        const { max, time } = this.props
        const interval = 120
        const step = (max - this.state.progress) * interval / (time * 1000)
        if (step <= 0) {
            this.setState({ progress: max })
        } else {
            const timeId = setInterval(() => {
                const progress = this.state.progress
                if (progress >= max) {
                    clearInterval(timeId)
                } else {
                    this.setState({ progress: this.state.progress + step })
                }
            }, interval)
        }
    }
    public toEnd() {
        this.setState({ progress: 100 })
        const onEnd = this.props.onEnd
        if (onEnd) {
            setTimeout(() => {
                onEnd()
            }, 1200)
        }
    }
    public render() {
        return (<Progress className={this.props.className}
            status='active'
            showInfo={false}
            strokeColor={{
                '0%': '#c8af8d',
                '100%': '#ffa81e',
            }}
            percent={this.state.progress}
            type="line"
            strokeWidth={3}
        />)
    }
}

export default ProgressBar
