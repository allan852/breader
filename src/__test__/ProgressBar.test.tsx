import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import EnzymeToJson from 'enzyme-to-json'
import React from 'react'
import ProgressBar, { IProgressBarProps } from '../components/ProgressBar'

Enzyme.configure({ adapter: new EnzymeAdapter() })

describe('ProgressBar Testing', () => {
    const propsMock: IProgressBarProps = {
        max: 80,
        onEnd: jest.fn(),
        time: 2,
    }

    it('dom testing', () => {
        const wrapper = Enzyme.mount(
            <ProgressBar {...propsMock} />
        )
    })

    it('snapshot testing', () => {
        const wrapper = Enzyme.mount(<ProgressBar {...propsMock} />)
        const tree = EnzymeToJson(wrapper)
        expect(tree).toMatchSnapshot()
    })
})
