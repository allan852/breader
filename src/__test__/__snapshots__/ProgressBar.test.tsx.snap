// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Testing snapshot testing 1`] = `
<ProgressBar
  max={80}
  onEnd={[MockFunction]}
  time={2}
>
  <Progress
    gapDegree={0}
    percent={0}
    showInfo={false}
    size="default"
    status="active"
    strokeColor={
      Object {
        "0%": "#c8af8d",
        "100%": "#ffa81e",
      }
    }
    strokeLinecap="round"
    strokeWidth={3}
    trailColor="#f3f3f3"
    type="line"
  >
    <div
      className="ant-progress ant-progress-line ant-progress-status-active ant-progress-default"
    >
      <Line
        gapDegree={0}
        percent={0}
        prefixCls="ant-progress"
        showInfo={false}
        size="default"
        status="active"
        strokeColor={
          Object {
            "0%": "#c8af8d",
            "100%": "#ffa81e",
          }
        }
        strokeLinecap="round"
        strokeWidth={3}
        trailColor="#f3f3f3"
        type="line"
      >
        <div>
          <div
            className="ant-progress-outer"
          >
            <div
              className="ant-progress-inner"
            >
              <div
                className="ant-progress-bg"
                style={
                  Object {
                    "backgroundImage": "linear-gradient(to right, #c8af8d 0%, #ffa81e 100%)",
                    "borderRadius": "100px",
                    "height": 3,
                    "width": "0%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </Line>
    </div>
  </Progress>
</ProgressBar>
`;
